File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array){
        int answer = 0;
        int[] ary = new int[1001];
  
    for(int i=0; i<array.length; i++){
        ary[array[i]]++;
  }
    int max = 0;
    int max_index = 0;
    
    for(int i =0; i<ary.length; i++){
        if(ary[i]>max){
            max = ary[i];
            max_index = i;
        }
    }
    int count = 0;
    for(int i =0; i<ary.length; i++){
        if(ary[i] == max)
            count++;
    }
    
    if(count > 1)
        return -1;
    answer = max_index;
    return answer;
    }
}   

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> arr(2010);

int solution(vector<int> array) {
    int answer = 0;
    for(int i = 0; i < array.size(); i++){
        arr[array[i]]++;
    }

    int max = 0;
    for(int i = 0; i < arr.size(); i++){
        if(max < arr[i])
            max = arr[i];
    }

    int cnt = 0;
    for(int i = 0; i < arr.size(); i++){
        if(max == arr[i]){
            answer = i;
            cnt++;
            if(cnt > 1)
                return -1;
        }
    }

    return answer;
}